user/manager.go

```diff
@@ -1505,6 +1578,67 @@ func (a *Manager) Close() error {
 	return a.db.Close()
 }
 
+func (a *Manager) maybeProvisionUsersAndAccess() error {
+	if !a.config.ProvisionEnabled {
+		return nil
+	}
+	users, err := a.Users()
+	if err != nil {
+		return err
+	}
+	provisionUsernames := util.Map(a.config.ProvisionUsers, func(u *User) string {
+		return u.Name
+	})
+	return execTx(a.db, func(tx *sql.Tx) error {
+		// Remove users that are provisioned, but not in the config anymore
+		for _, user := range users {
+			if user.Name == Everyone {
+				continue
+			} else if user.Provisioned && !util.Contains(provisionUsernames, user.Name) {
+				log.Tag(tag).Info("Removing previously provisioned user %s", user.Name)
+				if err := a.removeUserTx(tx, user.Name); err != nil {
+					return fmt.Errorf("failed to remove provisioned user %s: %v", user.Name, err)
+				}
+			}
+		}
+		// Add or update provisioned users
+		for _, user := range a.config.ProvisionUsers {
+			if user.Name == Everyone {
+				continue
+			}
+			existingUser, exists := util.Find(users, func(u *User) bool {
+				return u.Name == user.Name
+			})
+			if !exists {
+				log.Tag(tag).Info("Adding provisioned user %s", user.Name)
+				if err := a.addUserTx(tx, user.Name, user.Hash, user.Role, true, true); err != nil && !errors.Is(err, ErrUserExists) {
+					return fmt.Errorf("failed to add provisioned user %s: %v", user.Name, err)
+				}
+			} else if existingUser.Hash != user.Hash || existingUser.Role != user.Role {
```

Comment:
You're searching through all existing users in the DB, not just provisioned users. So it seems like this could change the password or role of a normal (non-provisioned) user. Is that intended?

Reply:
Im gonna take a look. Good catch!